### PR TITLE
🧹 Project Cleanup and Bundling

### LINKED ISSUE #70: ✨  Allow Bundling for low level projects
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

Simplify build chain and ease the effort on linting the projects. Allow a simple bundle to ensure the `api-client` and `library` can be exported to projects outside of electionguard-ui

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by keithrfung**
[//]: # (🚨 Please review the CONTRIBUTING.md in this repository. 💔Thank you!)

### Issue
*Link your PR to an issue*

Fixes #70

### Description
This is fairly large pull request so read through by commits. 
The biggest part is ensuring we use bundling and creating a new api-client project.
There is also heavy cleanup to reduce the commands simplify everything. 

### Testing
`make install`
`make lint`
`make build`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+
+# dependencies
+node_modules
+.pnp
+.pnp.js
+
+# dependencies
+node_modules
+.pnp
+.pnp.js
```
Comment (SteveMaier-IRT) [inline .gitignore:17]: Not sure why these are duplicated

File: .gitignore [hunk 2]
```diff
@@ -1,10 +1,34 @@
-# ignore all build folders
-**/build/
-**/node_modules/
-**/storybook-static/
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+
+# build
+build
```
Comment (SteveMaier-IRT) [inline .gitignore:4]: the ** will make sure that it matches any subfolder build, etc.  I think the way you have it now will just ignore the top level folders.

File: packages/admin-app/tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
+        "esModuleInterop": true,
+        "skipLibCheck": true,
+        "allowSyntheticDefaultImports": true,
+        "strict": true,
+        "forceConsistentCasingInFileNames": true,
+        "noFallthroughCasesInSwitch": true,
+        "resolveJsonModule": true,
+        "isolatedModules": true,
+        "noEmit": true,
+        "target": "es5"
```
Comment (SteveMaier-IRT) [inline packages/admin-app/tsconfig.json:17]: For both React projects, the target is set to es5, but in the lib section esnext is used.  Are these being different expected?

File: .gitignore (hunk 1 above)
Comment (keithrfung) [reply .gitignore:17]: Mistake. Will fix.

File: .gitignore (hunk 2 above)
Comment (keithrfung) [reply .gitignore:4]: Tested. With a flat name it will get both.

Comment (keithrfung) [reply .gitignore:4]: Screen Shot 2021-10-05 at 8 25 36 AM

File: packages/result-app/package.json [hunk 4]
```diff
@@ -10,23 +14,21 @@
         "@types/node": "^12.0.0",
         "@types/react": "^17.0.0",
         "@types/react-dom": "^17.0.0",
```
Comment (SteveMaier-IRT) [inline packages/result-app/package.json:16]: version 17.0.9 was released 3 months ago.  When react was updated to 17.0.2 6 months ago, these also were updated to 17.0.2/3

Comment (keithrfung) [reply packages/result-app/package.json:17]: I don't see what you're saying

**Review by SteveMaier-IRT — APPROVED** [review]

**Review by SteveMaier-IRT — APPROVED** [review]

**Pull request merged**